azurerm/tags.go

```diff
@@ -79,16 +84,24 @@ func expandTags(tagsMap map[string]interface{}) map[string]*string {
 	return output
 }
 
-func flattenAndSetTags(d *schema.ResourceData, tagsMap map[string]*string) {
+func flattenAndSetTags(d *schema.ResourceData, tagsMap map[string]*string, skipPropNames ...string) {
 	if tagsMap == nil {
 		d.Set("tags", make(map[string]interface{}))
 		return
 	}
 
 	output := make(map[string]interface{}, len(tagsMap))
 
+	// Only first optional parameter will be processed.
+	skipPropName := ""
+	if len(skipPropNames) > 0 && len(skipPropNames[0]) > 0 {
+		skipPropName = skipPropNames[0]
+	}
+
 	for i, v := range tagsMap {
-		output[i] = *v
+		if !strings.EqualFold(i, skipPropName) {
```

Comment:
Why using case insensitive comparison instead of "==" operator here?

Reply:
"$type" seems to be reserved usage in PLAN, it should be case insensitive. Or, to be safer, we should use case sensitive instead.

Reply:
👍 for it being case insensitive